tests/h/views/accounts_test.py

```diff
@@ -660,6 +660,29 @@ def test_get_sets_subscriptions_data_in_form(self,
             'notifications': set(['reply']),
         })
 
+    def test_get_for_user_without_email_address(self,
+                                                form_validating_to,
+                                                pyramid_request):
+        controller = views.NotificationsController(pyramid_request)
+        controller.form = form_validating_to({})
+
+        result = controller.get()
+
+        assert 'form' not in result
```

Comment:
I'm _guessing_ that these results happen because there is no `request.user`? But I have to guess that. It's also not a common (or even possible?) state for this form—you have to be logged in to get here, right?

Perhaps use `factories` to create a `User` object with email set to `None`, and then set `request.user` to that user? Let me know if I'm misunderstanding the intent of this test!

Reply:
You are right. The reason `user_has_email_address` is None is here is because `request.user` is. I can make it more explicit, good idea.